lib/ice_cube/validations/monthly_by_set_pos.rb

```diff
@@ -0,0 +1,85 @@
+module IceCube
+
+  module Validations::MonthlyBySetPos
+
+    def by_set_pos(*by_set_pos)
+      return by_set_pos([by_set_pos]) if by_set_pos.is_a?(Integer)
+
+      unless by_set_pos.nil? || by_set_pos.is_a?(Array)
+        raise ArgumentError, "Expecting Array or nil value for count, got #{by_set_pos.inspect}"
+      end
+      by_set_pos.flatten!
+      by_set_pos.each do |set_pos|
+        unless (set_pos >= -366 && set_pos <= -1) ||
+            (set_pos <= 366 && set_pos >= 1)
+          raise ArgumentError, "Expecting number in [-366, -1] or [1, 366], got #{set_pos} (#{by_set_pos})"
```

Comment:
Should this error display only the invalid `set_pos`, instead of the complete `by_set_post` list?